src/resolve/resolved-conf.c

```diff
@@ -299,6 +299,37 @@ int config_parse_dnssd_service_type(
         return 0;
 }
 
+int config_parse_dnssd_service_subtype(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        DnssdService *s = ASSERT_PTR(userdata);
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                s->type = mfree(s->subtype);
+                return 0;
+        }
+
+        if (!dns_subtype_name_is_valid(rvalue)) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "Service subtype is invalid. Ignoring.");
+                return 0;
+        }
+
+        return free_and_strdup_warn(&s->subtype, rvalue);
+}
```

Comment:
I'm not sure if it's related to this PR but with `SubType=` I got
```sh
=================================================================
==241773==ERROR: LeakSanitizer: detected memory leaks
Direct leak of 11 byte(s) in 1 object(s) allocated from:
    #0 0x7fd82dcd4e91 in __interceptor_strdup (/usr/lib64/clang/16/lib/libclang_rt.asan-x86_64.so+0xd4e91) (BuildId: 2a39bb4aa6d13ff866a4a47997cefdbe31a0350f)
    #1 0x7fd82ce3ef56 in free_and_strdup /home/vagrant/systemd-asan/build/../src/basic/string-util.c:1092:21
    #2 0x43d444 in config_parse_dnssd_service_type /home/vagrant/systemd-asan/build/../src/resolve/resolved-conf.c:295:13
    #3 0x7fd82c484fea in next_assignment /home/vagrant/systemd-asan/build/../src/shared/conf-parser.c:156:24
    #4 0x7fd82c469243 in parse_line /home/vagrant/systemd-asan/build/../src/shared/conf-parser.c:262:16
    #5 0x7fd82c4662d6 in config_parse /home/vagrant/systemd-asan/build/../src/shared/conf-parser.c:405:21
    #6 0x7fd82c46ba00 in config_parse_many_files /home/vagrant/systemd-asan/build/../src/shared/conf-parser.c:561:21
    #7 0x7fd82c46a132 in config_parse_many /home/vagrant/systemd-asan/build/../src/shared/conf-parser.c:623:13
    #8 0x5a339b in dnssd_service_load /home/vagrant/systemd-asan/build/../src/resolve/resolved-dnssd.c:92:13
    #9 0x5a1596 in dnssd_load /home/vagrant/systemd-asan/build/../src/resolve/resolved-dnssd.c:204:21
    #10 0x6042c5 in manager_new /home/vagrant/systemd-asan/build/../src/resolve/resolved-manager.c:629:13
    #11 0x683c55 in run /home/vagrant/systemd-asan/build/../src/resolve/resolved.c:72:13
    #12 0x68270d in main /home/vagrant/systemd-asan/build/../src/resolve/resolved.c:99:1
    #13 0x7fd82b249b89 in __libc_start_call_main (/lib64/libc.so.6+0x27b89) (BuildId: 70e92bb237883be3065a6afc9f0696aef2d068bf)
    #14 0x7fd82b249c4a in __libc_start_main@GLIBC_2.2.5 (/lib64/libc.so.6+0x27c4a) (BuildId: 70e92bb237883be3065a6afc9f0696aef2d068bf)
    #15 0x40d534 in _start (/home/vagrant/systemd-asan/build/systemd-resolved+0x40d534) (BuildId: 331b141e8a7288fe8d80eb125eff50797c791e17)
SUMMARY: AddressSanitizer: 11 byte(s) leaked in 1 allocation(s).
```

Reply:
Pasta error.